ansible/roles/dashmate/tasks/main.yml

```diff
@@ -148,30 +149,113 @@
   changed_when: true
   when: dashmate_branch is defined
 
-# TODO: we should use status command to figure out if dashmate is running
-#   to get the status from the response
-# Restart dashmate if already running
-- name: Restart dashmate services
-  ansible.builtin.command: "{{ dashmate_cmd }} restart --verbose"
+- name: Get status
+  ansible.builtin.command: "{{ dashmate_cmd }} status --format=json"
   become: true
   become_user: dashmate
+  changed_when: true
   args:
     chdir: '{{ dashmate_cwd }}'
-  register: dashmate_restart
-  changed_when: dashmate_restart.rc == 0
-  ignore_errors: true
+  register: dashmate_status
+
+- name: Get service status
+  ansible.builtin.set_fact:
+    core_docker_status: "{{ dashmate_status.stdout | from_json | json_query('core.dockerStatus') }}"
```

Comment:
What if tenderdash is stopped it Drive or DAPI is started?

Reply:
What do you mean? Should I rather check by drive container?

Reply:
If one of them started?

Reply:
I'm worried that it will add more fragility to the code. We actually don't have them stopped separately from each other. Tenderdash container will be either in `not_started` state or `running`. If there are any issues happening, it will remain in `restarting`, `exited`, or any other states, and tasks will be skipped